biz.aQute.repository/test/aQute/bnd/repository/maven/pom/provider/PomRepositoryTest.java

```diff
@@ -259,6 +264,58 @@ public void testBndPomRepoRefresh() throws Exception {
 		}
 	}
 
+	/**
+	 * Performs an update to the POM (testdata/pomrepo/simple.xml) to check if the
+	 * repo has been updated via polling. The changed content will be reset at the
+	 * end of the test. Furthermore a remote POM is added to the config to check run
+	 * through the stale-check
+	 */
+	public void testBndPomRepoFilePolling() throws Exception {
+		BndPomRepository bpr = new BndPomRepository();
+		Workspace w = Workspace.createStandaloneWorkspace(new Processor(), tmp.toURI());
+		w.setBase(tmp);
+		bpr.setRegistry(w);
+
+		Map<String,String> config = new HashMap<>();
+		config.put("pom",
+				"testdata/pomrepo/simple.xml"
+						+ ",https://repo1.maven.org/maven2/org/apache/felix/org.apache.felix.gogo.shell/0.12.0/org.apache.felix.gogo.shell-0.12.0.pom");
+		config.put("snapshotUrls", "https://repo1.maven.org/maven2/");
+		config.put("releaseUrls", "https://repo1.maven.org/maven2/");
+		config.put("name", "test");
+		config.put("pollTime", "100");
+		bpr.setProperties(config);
+
+		List<String> list = bpr.list(null);
+		assertNotNull(list);
+		assertEquals(2, list.size()); // one from simple.xml, two from the remote pom (but one is the same)
+
+		Path path = Paths.get("testdata/pomrepo/simple.xml");
+		byte[] originalContent = Files.readAllBytes(path);
+
+		List<String> lines = Files.readAllLines(path, StandardCharsets.UTF_8);
+
+		try {
+			StringBuilder sb = new StringBuilder();
+			for (String line : lines) {
+				if (line.contains("</dependencies>")) {
+					sb.append(
+							"<dependency><groupId>org.slf4j</groupId><artifactId>slf4j-api</artifactId><version>1.7.25</version></dependency>");
+				}
+				sb.append(line);
+			}
+			Files.write(path, sb.toString().getBytes(StandardCharsets.UTF_8), StandardOpenOption.WRITE,
```

Comment:
You should not write over the original file. If the test needs to change the file, first copy to a temp area and the mutate the temp file as needed for the test.